### PR TITLE
[PY-Bindings] add some doc to MSExperiment and MzMLFile

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Incremental doc improvement

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
```
[100%] Built target prepare_pyopenms_libs
Scanning dependencies of target pyopenms_create_cpp
Level 25:root:parsing progress 0 out of 530
Level 25:root:parsing progress 50 out of 530
Traceback (most recent call last):
  File "create_cpp_extension.py", line 63, in <module>
    decls, instance_map = autowrap.parse(pxd_files, ".", num_processes=int(PY_NUM_THREADS))
  File "/usr/local/lib/python2.7/dist-packages/autowrap/__init__.py", line 51, in parse
    return DeclResolver.resolve_decls_from_files(files, root, num_processes)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/DeclResolver.py", line 239, in resolve_decls_from_files
    return resolve_decls_from_files_single_thread(pathes, root)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/DeclResolver.py", line 232, in resolve_decls_from_files_single_thread
    decls.extend(PXDParser.parse_pxd_file(full_path))
  File "/usr/local/lib/python2.7/dist-packages/autowrap/PXDParser.py", line 518, in parse_pxd_file
    result.append(handler(body, lines, path))
  File "/usr/local/lib/python2.7/dist-packages/autowrap/PXDParser.py", line 315, in parseTree
    decl = MethodOrAttributeDecl.parseTree(att, lines, pxd_path)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/PXDParser.py", line 395, in parseTree
    annotations = parse_line_annotations(node, lines)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/PXDParser.py", line 125, in parse_line_annotations
    assert value.strip(), "empty value for key '%s'" % key
AssertionError: empty value for key 'wrap-doc'
src/pyOpenMS/CMakeFiles/pyopenms_create_cpp.dir/build.make:76: recipe for target 'src/pyOpenMS/CMakeFiles/pyopenms_create_cpp' failed
```

**Comment by jpfeuffer** [comment]
It has to be `wrap-doc:DSDASDASDAS` without space. Or a newline.
See https://github.com/uweschmitt/autowrap/blob/2dcf9724294f23f87ae686f7d5a5f193a5fab3e2/autowrap/PXDParser.py#L119

None of the autowrap Exceptions tell you where they occur. I think this would be a huge task.

**Comment by timosachsenberg** [comment]
but 
- one could output the line where the error occurred
- point out that the likely cause is a space after :

**Comment by jpfeuffer** [comment]
Yes sure. But there are hundreds of different functions in autowrap that would require better error/exception handling. Not only parse_line_annotation.

File: src/pyOpenMS/pxds/MSExperiment.pxd [hunk 1]
```diff
@@ -85,19 +85,19 @@ cdef extern from "<OpenMS/KERNEL/MSExperiment.h>" namespace "OpenMS":
...
         void reserve(Size s) nogil except +
-        Size getNrSpectra() nogil except +
-        Size getNrChromatograms() nogil except +
+        Size getNrSpectra() nogil except + # wrap-doc:return the number of MS spectra
+        Size getNrChromatograms() nogil except + # wrap-doc:returns the number of chromatograms
         libcpp_vector[unsigned int] getMSLevels() nogil except +  # wrap-ignore
 
-        void sortSpectra(bool sort_mz) nogil except +
+        void sortSpectra(bool sort_mz) nogil except + # wrap-doc:Sort spectra by RT. If sort_mz=True also sort each peak in a spectrum by m/z
         void sortSpectra() nogil except +
```
Comment (JeeH-K) [inline src/pyOpenMS/pxds/MSExperiment.pxd:93]: add comments to annotate the difference between sortSpectra(bool sort_mz) and sortSpectra() ?

File: src/pyOpenMS/pxds/MSSpectrum.pxd [hunk 2]
```diff
@@ -49,32 +88,32 @@ cdef extern from "<OpenMS/KERNEL/MSSpectrum.h>" namespace "OpenMS":
...
 
-        bool isSorted() nogil except +
+        bool isSorted() nogil except + #wrap-doc:Returns true if the spectrum is sorte by m/z
 
-        int findNearest(double) nogil except+
-        int findNearest(double, double) nogil except+
-        int findNearest(double, double, double) nogil except+
-        int findHighestInWindow(double, double, double) nogil except+
+        int findNearest(double) nogil except + #wrap-doc:Returns the index of the closest peak in m/z
+        int findNearest(double, double) nogil except + #wrap-doc:Returns the index of the closest peak in the provided +/- m/z tolerance window (-1 if none match)
```
Comment (JeeH-K) [inline src/pyOpenMS/pxds/MSSpectrum.pxd:94]: adding an exact description of parameters may help readers
i.e) params=(double mz, double tolerance)

Comment (timosachsenberg) [reply src/pyOpenMS/pxds/MSSpectrum.pxd:94]: good point!

File: src/pyOpenMS/pxds/MSExperiment.pxd (hunk 1 above)
Comment (timosachsenberg) [reply src/pyOpenMS/pxds/MSExperiment.pxd:93]: I think it is just an overload because cython doesn't support optional parameters

**Pull request merged**